SOL-1182 Fix of Flaky test in proctored exams.

muhhshoaib opened the pull request.

muhhshoaib commented:
jenkins run js

muhhshoaib commented:
jenkins run bokchoy

muhhshoaib commented:
jenkins run bokchoy

muhhshoaib commented on common/test/acceptance/tests/lms/test_lms_instructor_dashboard.py:
```diff
@@ -222,14 +199,13 @@ def _create_a_timed_exam_and_attempt(self):
-        # Given that an exam has been configured to be a proctored exam.
-        self._create_a_proctored_exam_and_attempt()
+        # Given that an exam has been configured to be a timed exam.
+        self._create_a_timed_exam_and_attempt()
```
No need to create a proctored exam attempt in this test case. I removed the proctored exam attempt method `_create_a_proctored_exam_and_attempt ` and used the `create_a_timed_exam_and_attempt ` method to create the attempt.

muhhshoaib replied:
All the other test use `create_a_timed_exam_and_attempt` and they are passing without any flakiness.

muhhshoaib commented on common/test/acceptance/tests/lms/test_lms_instructor_dashboard.py:
```diff
@@ -222,14 +199,13 @@ def _create_a_timed_exam_and_attempt(self):
         self.courseware_page.start_timed_exam()
 
-    @flaky  # TODO fix this, see SOL-1183
+    @flaky(max_runs=10, min_passes=10)
```
NOTE: I have added the flaky decorator to pass the `max_run` and `min_passes` values to 10 and the test cases are passing now. I'll remove this once reviewed.

muhhshoaib commented:
@chrisndodge want to take a look into changes for the flaky test

chrisndodge commented on common/test/acceptance/tests/lms/test_lms_instructor_dashboard.py:
```diff
@@ -176,29 +176,6 @@ def _auto_auth(self, username, email, staff, enrollment_mode="honor"):
-
-        # Start the proctored exam.
-        self.courseware_page.start_proctored_exam()
-
```
@muhhshoaib even though we might not be calling this helper method, it seems like it'd be useful to keep around for future use. Is there any particular reason why we should delete it except for remove uncalled code?

muhhshoaib replied:
@chrisndodge we will keep this method for future use. I'll revert back this method.

chrisndodge commented:
@benpatterson @jzoldak any interest in checking this out? I guess next step is to remove the flakey decorator, right?

muhhshoaib commented:
@chrisndodge yes I'll remove this once it is reviewed.

chrisndodge commented:
+1

@muhhshoaib thanks for the fix. Please remove the flakey decorator, squash commits, and merge. If this proves to still be flakey, we'll re-open it.

chrisndodge commented:
sorry, @muhhshoaib, just to be maybe a bit clearer after pushing squashed changes, please wait for a green build before merging. Thx.

muhhshoaib commented:
@chrisndodge sure.

muhhshoaib commented:
jenkins run bokchoy

benpatterson commented:
@muhhshoaib to be sure it's not flaky, I recommend running it many times locally on chrome. There are instructions on that under "Tips for fixing flaky tests" on [this page](https://openedx.atlassian.net/wiki/display/TE/Flaky+Test+Process).

muhhshoaib commented:
@benpatterson I had used the flaky decorator in the previous build and pass the `max_runs=10` and `min_passes='10'`. After the build had passed using that decorator. Now i have squashed the commits and removed the decorator.

benpatterson commented:
@muhhshoaib that's great. Thanks.

muhhshoaib commented:
jenkins run python

jzoldak commented:
nice @muhhshoaib LGTM :+1:
